cartographer_rviz/cartographer_rviz/submaps_display.cc

```diff
@@ -149,6 +162,53 @@ void SubmapsDisplay::update(const float wall_dt, const float ros_dt) {
   }
 }
 
+void SubmapsDisplay::AllEnabledChanged() {
+  ::cartographer::common::MutexLocker locker(&mutex_);
+  bool visibility = visibility_all_enabled_->getBool();
+  for (auto &trajectory : trajectories_) {
+    for (auto &submap : trajectory) {
+      DisconnectVisibilityChange(submap);
+      submap->SetVisibility(visibility);
+      ConnectVisibilityChange(submap);
+    }
+  }
+}
+
+void SubmapsDisplay::VisibilityChanged(DrawableSubmap* submap) {
```

Comment:
In those cases, I don't think it matters since the macros are hiding the implementation. Here, you're never changing the value of the submap argument. So, it can be const. Not a const pointer, but a const arg.